### ISSUE TITLE
Textures

### ISSUE THREAD

**Issue by bomanviktor** [bomanviktor/rt#13]
Different textures can be added.

These include:

- [ ] Metal (reflective)
- [ ] Glass (refractive)
- [ ] Glossy (mix of diffuse and reflective)
- [ ] Textured (add image to the surface, very optional for us)

Before we start working on this, we should refactor the way we store colors and textures.

My suggestion is the super overpowered `enum` that rust has.

An implementation could look as follows:

```rust
    pub enum Texture<T> {
        Diffusive(Vector3<f64>, T),
        Metal(Vector3<f64>, T),
        Glossy(Vector3<f64>, T),
        Glass(Vector3<f64>, T),
        Light(Vector3<f64>, T),
    }   
```

T could be whatever we want, and the Vector3<f64> is the color of the object.

T for Light could for example be brightness. T for diffuse could be amount of secondary rays. T for metal could be how reflective it is etc.

Then we are able to send both the color and any other property into their respective function. I have already implemented this on a separate branch and will push this, then if you want to add one texture just assign yourself here and make a branch (and merge into the textures branch).

This is more of a bonus thing obviously but if you wanna do a different implementation of the current logic then this is a perfect start.
